src/ast/passes/codegen_llvm.cpp

```diff
@@ -2658,9 +2658,14 @@ AllocaInst *CodegenLLVM::getMultiMapKey(Map &map,
                                      key,
```

Comment:
IIRC CreateGEP creates the right type based on its arguments so if we have to cast `offset_val` the type of `key_type` must be wrong?

Reply:
IIUC, `key_type` is a byte array (to comprise all possible key types) so we have to cast the result of the GEP to the type that we want to store. The question here is if the cast to `i64 *` is correct, as the extra key may not always be an `i64` (although currently it always is).

Btw. there's a bug in this loop - `offset` is never incremented so it won't work with multiple `extra_keys` (at this point, at most one extra key is passed). @lenticularis39 perhaps you could fix that as a part of this PR?

Reply:
The extra key is currently only used in histogram maps, where it is always `i64`. I guess we can cast to `extra_key->getType()->getPointerTo()` to accommodate for possible future uses of extra key.

As of the missing `offset` incrementation: the extra keys are passed as a vector of `Value`s, whose type has to be inferred from `DataLayout` (unlike the type of non-extra keys). `extra_keys_size` only contains the sum of the sizes of the extra keys (which is now the same as the size of the extra key, since there is only one used). Do you think it would be better to pass a vector of sizes instead of `extra_keys_size` or to drop the argument altogether and infer the sizes from `DataLayout`?

Reply:
Let's infer it from data layout, if possible.

Reply:
OK, I pushed both changes into the PR branch.